plugins/terminal.koplugin/main.lua

```diff
@@ -1,57 +1,402 @@
     self.ui.menu:registerToMainMenu(self)
+    self.shortcuts = self.settings:readSetting("shortcuts") or {}
+end
+
+function Terminal:saveShortcuts()
+    self.settings:saveSetting("shortcuts", self.shortcuts)
+    self.settings:flush()
+    UIManager:show(InfoMessage:new{
+        text = _("Shortcuts saved"),
+        timeout = 2
+    })
+end
+
+function Terminal:manageShortcuts()
+    self.shortcuts_dialog = CenterContainer:new {
+        dimen = Screen:getSize(),
+    }
+    self.shortcuts_menu = Menu:new{
+        show_parent = self.ui,
+        width = Screen:getWidth(),
+        height = Screen:getHeight(),
+        covers_fullscreen = true, -- hint for UIManager:_repaint()
+        is_borderless = true,
+        is_popout = false,
+        perpage = self.items_per_page,
+        onMenuHold = self.onMenuHoldShortcuts,
+        _manager = self,
+    }
+    table.insert(self.shortcuts_dialog, self.shortcuts_menu)
+    self.shortcuts_menu.close_callback = function()
+        UIManager:close(self.shortcuts_dialog)
+    end
+
+    -- sort the shortcuts:
+    if #self.shortcuts > 0 then
+        table.sort(self.shortcuts, function(v1, v2)
+            return v1.text < v2.text
+        end)
+    end
+    self:updateItemTable()
+end
+
+function Terminal:updateItemTable()
+    local item_table = {}
+    if #self.shortcuts > 0 then
+        local actions_count = 3 -- separator + actions
+        for nr, f in ipairs(self.shortcuts) do
+            local item = {
+                nr = nr,
+                text = f.text,
+                commands = f.commands,
+                editable = true,
+                deletable = true,
+                callback = function()
+                    -- so we know which middle button to display in the results:
+                    self.source = "shortcut"
+                    -- execute immediately, skip terminal dialog:
+                    self.command = self:ensureWhitelineAfterCommands(f.commands)
+                    Trapper:wrap(function()
+                        self:execute()
+                    end)
+                end
+            }
+            table.insert(item_table, item)
+            -- add page actions at end of each page with shortcuts:
+            local factor = self.items_per_page - actions_count
+            if nr % factor == 0 or nr == #self.shortcuts then
+                -- insert "separator":
+                table.insert(item_table, {
+                    text = " ",
+                    deletable = false,
+                    editable = false,
+                    callback = function()
+                        self:manageShortcuts()
+                    end,
+                })
+                -- actions:
+                self:insertPageActions(item_table)
+            end
+        end
+        -- no shortcuts defined yet:
+    else
+        self:insertPageActions(item_table)
+    end
+    local title = #self.shortcuts == 1 and _("Terminal shortcut") or _("Terminal shortcuts")
```

Comment:
this should probably use http://koreader.rocks/doc/modules/gettext.html#ngettext
```suggestion
    local title = N(_("Terminal shortcut"), _("Terminal shortcuts"), #self.shortcuts)
```
with this at the top
local N_ = _.ngettext